### DISCUSSION TITLE
Generate memory access traces?

### DISCUSSION THREAD

**Discussion by hkarl** [mortbopet/Ripes#374] [Ideas]
Hi, 

it might be interesting to use Ripes to generate memory access traces for concrete programs. It looks like all the necessary infrastructure to do so is in place; all that is needed would be a new CLI switch with a file name to which to write assembly instructions with content of corresponding registers / addresses actually used by a program. 

Then, these traces could be useful to drive, e.g., simulations of different caching algorithms, virtual memory schemes, ... 

It might be a considerable improvement and small overheads. 

Best, 
HK

**Comment by mortbopet** [comment]
I like the idea! PRs are always welcome 🚀 sounds like a good first project, as well, to get into the codebase.
